assets/common/scripts/game_manager.gd

```diff
@@ -42,47 +30,52 @@ func _shortcut_input(event):
 		increment_day()
 		get_viewport().set_input_as_handled()
 
+#region Zones
+var current_zone: Zone
+
 func register_zone(zone: Zone):
 	current_zone = zone
-	if not Savegame.player.zones.has(zone.id):
-		Savegame.player.zones[zone.id] = init_map()
+	if not Savegame.player.grid.has(zone.id):
+		Savegame.player.grid[zone.id] = init_map()
 	current_zone_updated.emit()
 
 func deregister_zone(zone: Zone):
 	if zone == current_zone:
 		current_zone = null
 		current_zone_updated.emit()
+#endregion
+
+#region Grid
+func get_grid():
+	var grid = Savegame.player.grid.get(current_zone.id)
+	if not grid:
+		Savegame.player.grid[current_zone.id] = {}
+		grid = Savegame.player.grid[current_zone.id]
+	return grid
+	
+var scanner_prop: String:
+	set(prop):
+		scanner_prop = prop
+		scanner_prop_updated.emit(prop)
 
 func get_grid_point(pos: Vector3) -> Dictionary:
 	var point = current_zone.grid.get_cell_by_position(pos)
 	return Savegame.player.area_map[point]
 
 func update_grid_property(center: Vector2i, property: String, radius: int, change: float):
-	if not CELL_PROPERTIES.has(property):
+	if not grid_props_dt.has(property):
 		Utils.log_error("Grid", "Grid does not have property ", property)
 		return
 	for x in range(center.x - radius, center.x + radius + 1):
 		for y in range(center.y - radius, center.y + radius + 1):
 			var point = Vector2i(x,y)
 			var dist = Vector2(point).distance_to(center)
 			var scaled_change = change * (radius - dist) / radius # scale down over distance
-			var zone = Savegame.player.zones[current_zone.id]
+			var zone = GameManager.get_grid()
 			if dist <= radius and zone.has(point):
 				zone[point][property] = clampf(zone[point][property] + scaled_change, 0, 1)
 	grid_updated.emit()
 
-func increment_day():
-	Savegame.player.day += 1
-	for zone_id in Savegame.player.crops:
-		for crop_cell in Savegame.player.crops[zone_id]:
-			var crop_entry = Savegame.player.crops[zone_id][crop_cell]
-			var crop_details: CropRow = GameManager.crops_dt.get_row(crop_entry.seed_id)
-			crop_entry.days_planted += 1
-			crop_entry.growth_score += 20
-			GameManager.update_grid_property(crop_cell, 'hydration', crop_details.effect_radius, -0.2)
-	Savegame.save_file()
-	day_incremented.emit()
-
 func init_map() -> Dictionary:
```

Comment:
Function should take in the zone as an argument, maybe also call it `init_grid_data` to be more clear as to what it does.